doc/dockerfile.rst

```diff
@@ -0,0 +1,139 @@
+Using a Dockerfile
+------------------
+
+Binder supports configuration files for package
+installation, environment specification, post-build shell scripts, and more.
+It should be possible to create the environment that you want *without*
+using a Dockerfile. For more information about the different environment
+configuration files that Binder can create, see the
+`guide to creating Binder-ready repositories <LINK>`_.
+
+However, in case you cannot meet all your needs with these configuration
+files, it is also possible to use a Dockerfile to define your environment.
+This is considered an **advanced use case**, and we cannot guarantee that the
+Dockerfile will work.
+
+This guide will help you in preparing your Dockerfile so that it has the
+components needed to run JupyterHub, allowing it to work on Binder
+deployments.
+
+.. note::
+
+   Binder's requirements for Dockerfiles are in beta and changing, so
+   Dockerfiles on Binder will break from time to time until Binder
+   solidifies its requirements. Dockerfiles are not an appropriate choice
+   if you require reproducibility with Binder at the moment.
+
+When should you use a Dockerfile?
+=================================
+
+Below are a few use-cases where you *might* want to use a Dockerfile with
+Binder.
+
+When you must inherit from a popular Docker image
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you want to use a pre-existing Docker image, you may source it in your
+Dockerfile. For example, this code sources the Jupyter-Scipy notebook:
+
+.. code-block:: Dockerfile
+
+    # Note that there must be a tag
+    FROM jupyter/scipy-notebook:cf6258237ff9
+
+See `Preparing your Dockerfile`_ for instructions on how to
+do this properly.
+
+When you are building complex software
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Some software is simply too complex to be specified with Binder configuration

```

Comment:
Most Binder configurations can be achieved without a Dockerfile. Before resorting to a Dockerfile, we recommend trying to use ``postBuild`` commands for configuration.